test/plugin/test_out_google_cloud_grpc.rb

```diff
@@ -205,31 +205,41 @@ def test_non_integer_timestamp
 
   private
 
-  GRPC_MOCK_HOST = 'localhost:56789'
-
   WriteLogEntriesRequest = Google::Logging::V2::WriteLogEntriesRequest
   WriteLogEntriesResponse = Google::Logging::V2::WriteLogEntriesResponse
 
   USE_GRPC_CONFIG = %(
     use_grpc true
   )
 
+  def generate_mock_host
+    "localhost:#{Random.new.rand(50_000..60_000)}"
+  end
+
   # Create a Fluentd output test driver with the Google Cloud Output plugin with
   # grpc enabled. The signature of this method is different between the grpc
   # path and the non-grpc path. For grpc, an additional grpc stub class can be
   # passed in to construct the mock used by the test driver.
   def create_driver(conf = APPLICATION_DEFAULT_CONFIG, tag = 'test',
                     grpc_stub = GRPCLoggingMockService.rpc_stub_class)
+    if @mock_host
+      mock_host = @mock_host
+      @mock_host = nil
+    else
+      mock_host = generate_mock_host
```

Comment:
Is this for cases where there are no stubs set up or this is the second driver within the stub context? Wouldn't we want to fail here instead?

Reply:
Some test creates the driver without setting up the logging stub.

Reply:
I added some comment to clarify this.

`@mock_host` only gets set up when `setup_logging_stubs` is called. Some tests create drivers without creating a logging stub context. For these cases just use a new mock port. 

Another option is to change these test cases. Yet this is a temporary workaround. I'd like to keep the size of the things we need to revert as small as possible.

Reply:
I'm confused — how do the tests succeed if the expectations are not set up? Does that mean that the expectations are irrelevant, because the tests never invoke the driver? Then there should be no need to set up the driver — it might as well be `nil`. What am I missing?

Reply:
For tests as simple as

```
  def test_configure_service_account_application_default
    setup_gce_metadata_stubs
    d = create_driver
    assert_equal HOSTNAME, d.instance.vm_name
  end
```

We never reach the point where a RPC call needs to be made. So logging stubs is never set up.

On a second thought, as long as we don't actually call the api_client in the driver, we don't require a mock host. So I just removed the logic to create a mock host for the driver when no logging stub was set up. It should be fine for it to stay `nil`.

PTAL.

Reply:
My point exactly.